sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -93,11 +93,49 @@ object ScalaNativePluginInternal {
                                      entry,
                                      verbose)
 
+      checkThatClangIsRecentEnough(clang)
+
       IO.createDirectory(target)
       unpackRtlib(classpath)
       compileNir(opts)
       compileLl(clang, target, appll, binary, clangOpts)
       Process(abs(binary)).!
     }
   )
-}
+
+  /**
+   * Tests whether the clang compiler is recent enough.
+   * <p>
+   * This is determined through looking up a built-in #define which is more reliable than testing
+   * for a specific version.
+   * <p>
+   * It might be better to use feature checking macros:
+   * http://clang.llvm.org/docs/LanguageExtensions.html#feature-checking-macros
+   */
+  private def checkThatClangIsRecentEnough(pathToClangBinary: File): Unit = {
+    val clang = maybeFile(pathToClangBinary)
+    val defines: Seq[String] = definesBuiltIn(clang).to[Seq].flatten
+    val clangIsRecentEnough = defines.contains("__DECIMAL_DIG__ __LDBL_DECIMAL_DIG__")
+
+    if (! clangIsRecentEnough) {
+      throw new MessageOnlyException(s"No recent installation of clang found at $pathToClangBinary.\nSee https://github.com/scala-native/scala-native/blob/master/docs/building.md for details.")
+    }
+  }
+
+  def maybeFile(f: File) = f match {
+    case clang if clang.exists => Some(abs(clang))
+    case none => None
+  }
+
+  private def definesBuiltIn(pathToClangBinary: Option[String]): Option[Seq[String]] = {
```

Comment:
Ok, moved it into `checkThatClangIsRecentEnough`.

I originally left it as a separate method so it can be used by other parts of the build. I assumed that querying clang's built-in defines was interesting for the build in general. Having it sit inside of another method makes it very hard to find (e.g. when using code completion in an IDE).